src/browser/api_protocol/transport_strategy/elipc_strategy.ts

```diff
@@ -35,15 +35,21 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
             if (strategyName !== this.constructor.name) {
                 next();
             } else if (typeof (action) === 'function') {
-                try {
-                    // singleFrameOnly check first so to prevent frame superceding when disabled.
-                    if (!data.singleFrameOnly === false || e.sender.isValidWithFrameConnect(e.frameRoutingId)) {
-                        action(identity, data, ack, nack);
-                    } else {
-                        nack('API access has been superseded by another frame in this window.');
-                    }
-                } catch (err) {
-                    nack(err);
+                // singleFrameOnly check first so to prevent frame superceding when disabled.
+                if (!data.singleFrameOnly === false || e.sender.isValidWithFrameConnect(e.frameRoutingId)) {
+                    Promise.resolve()
+                        .then(() => action(identity, data, ack, nack))
+                        .then(result => {
+                            if (result !== undefined) {
```

Comment:
No. Almost all API handlers (except for the new external application ones) do not return a value. Rather they internally bear the responsibility of Ack'ing. To do so here would cause the vast majority of functions to Ack twice.